Usagetest: simd feature

After you put the simd implementations between a featureflag I seem unable to import it to compile code with Simd types. "facet" does not have the feature, if i try to import facet-core directly:

```toml
facet = {git = "https://github.com/facet-rs/facet.git", rev = "0233867", features = ["chrono"]}
facet-json = {git = "https://github.com/facet-rs/facet.git", rev = "0233867"}
facet-core = {git = "https://github.com/facet-rs/facet.git", rev = "0233867", features = ["chrono", "simd"]}
facet-pretty = {git = "https://github.com/facet-rs/facet.git", rev = "0233867"}
facet-yaml = {git = "https://github.com/facet-rs/facet.git", rev = "0233867"}
facet-assert = {git = "https://github.com/facet-rs/facet.git", rev = "0233867"}
```


I still get:
```
ad\src\simd.rs:180:43: error[E0277]: the trait bound `Simd<f64, 8>: Facet<'_>` is not satisfied: the trait `Facet<'_>` is not implemented for `Simd<f64, 8>`
ad\src\simd.rs:12:43: error[E0277]: the trait bound `Simd<f32, 16>: Facet<'_>` is not satisfied: the trait `Facet<'_>` is not implemented for `Simd<f32, 16>`
```

--- a/facet/Cargo.toml
+++ b/facet/Cargo.toml
@@ -62,11 +62,18 @@ tuples-12 = ["facet-core/tuples-12"]
 # Provide Facet trait implementations for function pointers.
 fn-ptr = ["facet-core/fn-ptr"]
 
+# Provide Facet trait implementations for portable SIMD types (requires nightly).
+# This adds 84 type implementations which increases compile time.
+simd = ["facet-core/simd"]
+
 # Include doc comments in generated Shape/Field/Variant definitions
 doc = ["facet-macros/doc"]
 
 # Enable typo suggestions for unknown attributes/fields (uses strsim)
 helpful-derive = ["facet-macros/helpful-derive"]
+
+[build-dependencies]
+autocfg = "1.3"
 
 [dependencies]
 facet-core = { path = "../facet-core", version = "=0.31.8", default-features = false }

--- a/facet/build.rs
+++ b/facet/build.rs
@@ -1,0 +1,20 @@
+fn main() {
+    // Emit a cfg flag when portable_simd is available (nightly only for now).
+    println!("cargo::rustc-check-cfg=cfg(has_portable_simd)");
+
+    let ac = autocfg::new();
+    // Probe for portable_simd feature support (requires nightly).
+    if ac
+        .probe_raw(
+            r#"
+            #![feature(portable_simd)]
+            fn _test() { let _: core::simd::Simd<f32, 4>; }
+            "#,
+        )
+        .is_ok()
+    {
+        println!("cargo::rustc-cfg=has_portable_simd");
+    }
+
+    println!("cargo::rerun-if-changed=build.rs");
+}

--- a/facet/tests/simd.rs
+++ b/facet/tests/simd.rs
@@ -1,0 +1,30 @@
+// Only run these tests when the simd feature is enabled AND portable_simd is available
+#![cfg(all(feature = "simd", has_portable_simd))]
+#![feature(portable_simd)]
+
+use core::simd::Simd;
+use facet::{Def, Facet, Type, UserType};
+
+#[test]
+fn simd_f32x4_shape() {
+    let shape = <Simd<f32, 4> as Facet>::SHAPE;
+    assert_eq!(shape.type_identifier, "f32x4");
+    assert!(matches!(shape.ty, Type::User(UserType::Opaque)));
+    assert!(matches!(shape.def, Def::Scalar));
+}
+
+#[test]
+fn simd_f64x8_shape() {
+    let shape = <Simd<f64, 8> as Facet>::SHAPE;
+    assert_eq!(shape.type_identifier, "f64x8");
+    assert!(matches!(shape.ty, Type::User(UserType::Opaque)));
+    assert!(matches!(shape.def, Def::Scalar));
+}
+
+#[test]
+fn simd_f32x16_shape() {
+    let shape = <Simd<f32, 16> as Facet>::SHAPE;
+    assert_eq!(shape.type_identifier, "f32x16");
+    assert!(matches!(shape.ty, Type::User(UserType::Opaque)));
+    assert!(matches!(shape.def, Def::Scalar));
+}

